### PR TITLE
Optimize varint reading and writing for small ints.

### PR DESCRIPTION
**Pull request by robertwb**
This is used as a primitive for several other coders.

Before:

tiny_row, RowCoder, 1000 element(s)    : p. element median time cost: 7.22051e-07 sec, relative std: 12.45%
large_row, RowCoder, 1000 element(s)   : p. element median time cost: 3.12304e-06 sec, relative std: 5.01%
nullable_row, RowCoder, 1000 element(s): p. element median time cost: 3.27158e-06 sec, relative std: 6.49%
diverse_row, RowCoder, 1000 element(s) : p. element median time cost: 2.33555e-06 sec, relative std: 7.48%
small_int, FastPrimitivesCoder, 1000 element(s)                   : p. element median time cost: 1.66655e-07 sec, relative std: 9.18%
large_int, FastPrimitivesCoder, 1000 element(s)                   : p. element median time cost: 3.39985e-07 sec, relative std: 3.05%
small_int, LengthPrefixCoder[FastPrimitivesCoder], 1000 element(s): p. element median time cost: 5.70536e-07 sec, relative std: 4.27%
small_list, FastPrimitivesCoder, 1000 element(s)               : p. element median time cost: 1.57607e-06 sec, relative std: 3.05%
large_list, FastPrimitivesCoder, 1000 element(s)               : p. element median time cost: 4.76551e-05 sec, relative std: 4.31%
small_list, IterableCoder[FastPrimitivesCoder], 1000 element(s): p. element median time cost: 1.59943e-06 sec, relative std: 2.11%
large_list, IterableCoder[FastPrimitivesCoder], 1000 element(s): p. element median time cost: 4.66548e-05 sec, relative std: 3.59%

After:

tiny_row, RowCoder, 1000 element(s)    : p. element median time cost: 6.68406e-07 sec, relative std: 13.95%
large_row, RowCoder, 1000 element(s)   : p. element median time cost: 2.88701e-06 sec, relative std: 6.26%
nullable_row, RowCoder, 1000 element(s): p. element median time cost: 3.19052e-06 sec, relative std: 8.06%
diverse_row, RowCoder, 1000 element(s) : p. element median time cost: 2.24805e-06 sec, relative std: 6.15%
small_int, FastPrimitivesCoder, 1000 element(s)                   : p. element median time cost: 1.65582e-07 sec, relative std: 13.75%
large_int, FastPrimitivesCoder, 1000 element(s)                   : p. element median time cost: 3.14236e-07 sec, relative std: 6.83%
small_int, LengthPrefixCoder[FastPrimitivesCoder], 1000 element(s): p. element median time cost: 5.85556e-07 sec, relative std: 4.98%
small_list, FastPrimitivesCoder, 1000 element(s)               : p. element median time cost: 1.48404e-06 sec, relative std: 42.11%
large_list, FastPrimitivesCoder, 1000 element(s)               : p. element median time cost: 4.77622e-05 sec, relative std: 4.20%
small_list, IterableCoder[FastPrimitivesCoder], 1000 element(s): p. element median time cost: 1.50216e-06 sec, relative std: 1.90%
large_list, IterableCoder[FastPrimitivesCoder], 1000 element(s): p. element median time cost: 4.57406e-05 sec, relative std: 3.11%

------------------------

Thank you for your contribution! Follow this checklist to help us incorporate your contribution quickly and easily:

 - [ ] [**Choose reviewer(s)**](https://beam.apache.org/contribute/#make-your-change) and mention them in a comment (`R: @username`).
 - [ ] Mention the appropriate issue in your description (for example: `addresses #123`), if applicable. This will automatically add a link to the pull request in the issue. If you would like the issue to automatically close on merging the pull request, comment `fixes #<ISSUE NUMBER>` instead.
 - [ ] Update `CHANGES.md` with noteworthy changes.
 - [ ] If this contribution is large, please file an Apache [Individual Contributor License Agreement](https://www.apache.org/licenses/icla.pdf).

See the [Contributor Guide](https://beam.apache.org/contribute) for more tips on [how to make review process smoother](https://beam.apache.org/contribute/get-started-contributing/#make-the-reviewers-job-easier).

To check the build health, please visit [https://github.com/apache/beam/blob/master/.test-infra/BUILD_STATUS.md](https://github.com/apache/beam/blob/master/.test-infra/BUILD_STATUS.md)

GitHub Actions Tests Status (on master branch)
------------------------------------------------------------------------------------------------
[Build python source distribution and wheels](https://github.com/apache/beam/actions?query=workflow%3A%22Build+python+source+distribution+and+wheels%22+branch%3Amaster+event%3Aschedule)
[Python tests](https://github.com/apache/beam/actions?query=workflow%3A%22Python+Tests%22+branch%3Amaster+event%3Aschedule)
[Java tests](https://github.com/apache/beam/actions?query=workflow%3A%22Java+Tests%22+branch%3Amaster+event%3Aschedule)
[Go tests](https://github.com/apache/beam/actions?query=workflow%3A%22Go+tests%22+branch%3Amaster+event%3Aschedule)

See [CI.md](https://github.com/apache/beam/blob/master/CI.md) for more information about GitHub Actions CI.

### CODE REVIEW

**Comment by robertwb** [comment]
R: @anandinguva

File: sdks/python/apache_beam/coders/stream.pyx [hunk 1]
```diff
@@ -183,15 +192,16 @@ cdef class InputStream(object):
 
   cpdef libc.stdint.int64_t read_var_int64(self) except? -1:
     """Decode a variable-length encoded long from a stream."""
-    cdef long byte
+    # Inline common case.
+    cdef long byte = <unsigned char> self.allc[self.pos]
+    self.pos += 1
+    if byte < 0x7F:
```
Comment (AnandInguva) [inline sdks/python/apache_beam/coders/stream.pyx:198]: Can you explain this condition?

Comment (AnandInguva) [reply sdks/python/apache_beam/coders/stream.pyx:198]: Does this represent we need to read the next byte in the stream?

File: sdks/python/apache_beam/coders/stream.pyx [hunk 2]
```diff
@@ -59,6 +59,12 @@ cdef class OutputStream(object):
   cpdef write_var_int64(self, libc.stdint.int64_t signed_v):
     """Encode a long using variable-length encoding to a stream."""
     cdef libc.stdint.uint64_t v = signed_v
+    # Inline common case.
+    if v <= 0x7F and self.pos < self.buffer_size - 1:
```
Comment (AnandInguva) [inline sdks/python/apache_beam/coders/stream.pyx:63]: v < 0x7F ? Following this from line 198

File: sdks/python/apache_beam/coders/stream.pyx (hunk 1 above)
Comment (robertwb) [reply sdks/python/apache_beam/coders/stream.pyx:198]: Integers that fit into seven bits are stored as single bytes on the stream.

File: sdks/python/apache_beam/coders/stream.pyx (hunk 2 above)
Comment (robertwb) [reply sdks/python/apache_beam/coders/stream.pyx:63]: <= is correct, fixed above as well.

File: sdks/python/apache_beam/coders/stream.pyx (hunk 1 above)
Comment (AnandInguva) [reply sdks/python/apache_beam/coders/stream.pyx]: thanks

**Review by AnandInguva — APPROVED** [review]

**Pull request merged**